main.go

```diff
@@ -143,56 +147,117 @@ func main() {
 	// Setup the context that's going to be used in controllers and for the manager.
 	ctx := ctrl.SetupSignalHandler()
 
-	if err = (&controllers.GCPMachineReconciler{
+	if setupErr := setupReconcilers(ctx, mgr); setupErr != nil {
+		setupLog.Error(err, "unable to setup reconcilers")
+		os.Exit(1)
+	}
+
+	if setupErr := setupWebhooks(mgr); setupErr != nil {
+		setupLog.Error(err, "unable to setup webhooks")
+		os.Exit(1)
+	}
+
+	if setupErr := setupProbes(mgr); setupErr != nil {
+		setupLog.Error(err, "unable to setup probes")
+		os.Exit(1)
+	}
+
+	// +kubebuilder:scaffold:builder
+	setupLog.Info("starting manager", "version", version.Get().String(), "extended_info", version.Get())
+	if err := mgr.Start(ctx); err != nil {
+		setupLog.Error(err, "problem running manager")
+		os.Exit(1)
+	}
+}
+
+func setupReconcilers(ctx context.Context, mgr ctrl.Manager) error {
+	if err := (&controllers.GCPMachineReconciler{
 		Client:           mgr.GetClient(),
 		ReconcileTimeout: reconcileTimeout,
 		WatchFilterValue: watchFilterValue,
 	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpMachineConcurrency}); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "GCPMachine")
-		os.Exit(1)
+		return fmt.Errorf("setting up GCPMachine controller: %w", err)
 	}
-	if err = (&controllers.GCPClusterReconciler{
+	if err := (&controllers.GCPClusterReconciler{
 		Client:           mgr.GetClient(),
 		ReconcileTimeout: reconcileTimeout,
 		WatchFilterValue: watchFilterValue,
 	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpClusterConcurrency}); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "GCPCluster")
-		os.Exit(1)
+		return fmt.Errorf("setting up GCPCluster controller: %w", err)
 	}
 
-	if err = (&infrav1beta1.GCPCluster{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "GCPCluster")
-		os.Exit(1)
+	if feature.Gates.Enabled(feature.GKE) {
```

Comment:
We'll do this in the next PR when we bring it all together.

Reply:
reverted